src/main/java/supersymmetry/common/item/armor/JetWingpack.java

```diff
@@ -0,0 +1,325 @@
+package supersymmetry.common.item.armor;
+
+import gregtech.api.GTValues;
+import gregtech.api.capability.IFilter;
+import gregtech.api.capability.impl.GTFluidHandlerItemStack;
+import gregtech.api.items.armor.ArmorLogicSuite;
+import gregtech.api.items.armor.ArmorMetaItem;
+import gregtech.api.items.armor.ArmorUtils;
+import gregtech.api.items.metaitem.stats.*;
+import gregtech.api.recipes.Recipe;
+import gregtech.api.recipes.RecipeMaps;
+import gregtech.api.unification.material.Materials;
+import gregtech.api.util.GTUtility;
+import gregtech.api.util.GradientUtil;
+import gregtech.api.util.input.KeyBind;
+import net.minecraft.client.model.ModelBiped;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.*;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import supersymmetry.api.capability.IElytraFlyingProvider;
+import supersymmetry.api.capability.SuSyCapabilities;
+import supersymmetry.api.util.ElytraFlyingUtils;
+import supersymmetry.client.renderer.handler.JetWingpackModel;
+
+import java.awt.*;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+
+public class JetWingpack extends ArmorLogicSuite implements IItemHUDProvider {
+
+    public static final int TANK_CAPACITY = 32000;
+    public static final Function<FluidStack, Integer> COMBUSTION_FUEL_BURN_TIME = fluidStack -> {
```

Comment:
i'm thinking we should limit the fuels to only the kerosene and upgraded kerosenes since this runs on gas turbine engines.